Cabal/Distribution/PackageDescription/Configuration.hs

```diff
@@ -421,11 +421,14 @@ overallDependencies enabled (TargetSet targets) = mconcat depss
   where
     (depss, _) = unzip $ filter (removeDisabledSections . snd) targets
     removeDisabledSections :: PDTagged -> Bool
-    removeDisabledSections (Lib l)     = componentEnabled enabled (CLib l)
-    removeDisabledSections (SubLib _ l) = componentEnabled enabled (CLib l)
-    removeDisabledSections (Exe _ e)   = componentEnabled enabled (CExe e)
-    removeDisabledSections (Test _ t)  = componentEnabled enabled (CTest t)
-    removeDisabledSections (Bench _ b) = componentEnabled enabled (CBench b)
+    -- UGH. The embedded componentName in the 'Component's here is
+    -- BLANK.  I don't know whose fault this is but I'll use the tag
+    -- instead. -- ezyang
```

Comment:
I looked at this briefly, but I couldn't figure out how to convince the parser to set these properly. Maybe you might know? (I'm pretty sure this is straight out of the parser when it's building the cond tree)